### ISSUE TITLE
PWA crash in "Open links in incognito tabs always" mode

### ISSUE THREAD

**Issue by Elaborendum** [bromite/bromite#1116]
### Bromite version

Version: `90.0.4430.92`
Arch: `arm64`
Android version: `10`
Device model: `SM-A205G`

### Flags

These are the flags changed under `chrome://flags`: Reader Mode triggering

### Is this bug about the SystemWebView?

No

### Is the bug reproducible with latest version?

Yes

### Can the bug be reproduced with corresponding Chromium version?

No

### Is this bug happening in an incognito tab?
Yes

### Allow ads for the site; is the bug caused by the adblocker?
No

### Is the bug a crash?

Yes, this is the crash dump:
```
<I only have regular Windows at hand, so no grep, how do I do it?>
```

### Describe the bug

Trying to open a PWA while the "Open links in incognito tabs always" option is on will cause Bromite to crash.

I've seen other issues about this, but they were closed so I opened this one.

### Steps to reproduce the bug

Steps to reproduce the bug:
1. Activate "Open links in incognito tabs always" in settings.
2. Install a PWA
3. Open the PWA

### Expected behavior

PWA opens as it normally does in non incognito mode.

**Comment by uazo** [comment]
here the stacktrace

```
04-30 06:09:09.933 17452 17452 W System.err: java.lang.ClassCastException: org.chromium.chrome.browser.webapps.WebappIntentDataProvider cannot be cast to org.chromium.chrome.browser.customtabs.IncognitoCustomTabIntentDataProvider
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.customtabs.CustomTabActivityLifecycleUmaTracker.recordIncognitoLaunchReason(CustomTabActivityLifecycleUmaTracker.java:37)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.customtabs.CustomTabActivityLifecycleUmaTracker.recordMetrics(CustomTabActivityLifecycleUmaTracker.java:54)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.customtabs.CustomTabActivityLifecycleUmaTracker.onResumeWithNative(CustomTabActivityLifecycleUmaTracker.java:95)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.ActivityLifecycleDispatcherImpl.dispatchOnResumeWithNative(ActivityLifecycleDispatcherImpl.java:170)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.AsyncInitializationActivity.onResumeWithNative(AsyncInitializationActivity.java:544)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.app.ChromeActivity.onResumeWithNative(ChromeActivity.java:984)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.NativeInitializationController.onResume(NativeInitializationController.java:166)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.NativeInitializationController.onNativeInitializationComplete(NativeInitializationController.java:144)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.AsyncInitializationActivity.finishNativeInitialization(AsyncInitializationActivity.java:276)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.app.ChromeActivity.finishNativeInitialization(ChromeActivity.java:1470)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.customtabs.BaseCustomTabActivity.finishNativeInitialization(BaseCustomTabActivity.java:294)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.BrowserParts$$CC.startNativeInitialization$$dflt$$(Unknown Source:63)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.AsyncInitializationActivity.startNativeInitialization(Unknown Source:0)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.ChromeBrowserInitializer.lambda$handlePostNativeStartup$6$ChromeBrowserInitializer(ChromeBrowserInitializer.java:279)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.ChromeBrowserInitializer$$Lambda$7.run(Unknown Source:2)
04-30 06:09:09.933 17452 17452 W System.err:    at org.chromium.chrome.browser.init.ChainedTasks$1.run(ChainedTasks.java:37)
04-30 06:09:09.933 17452 17452 W System.err:    at android.os.MessageQueue.nativePollOnce(Native Method)
04-30 06:09:09.933 17452 17452 W System.err:    at android.os.MessageQueue.next(MessageQueue.java:326)
04-30 06:09:09.933 17452 17452 W System.err:    at android.os.Looper.loop(Looper.java:160)
04-30 06:09:09.933 17452 17452 W System.err:    at android.app.ActivityThread.main(ActivityThread.java:6669)
04-30 06:09:09.933 17452 17452 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
04-30 06:09:09.933 17452 17452 W System.err:    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
04-30 06:09:09.933 17452 17452 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
04-30 06:09:09.954 17452 17452 F chromium: [FATAL:jni_android.cc(306)] Please include Java exception stack in crash report
```

similar to https://github.com/bromite/bromite/issues/1051#issuecomment-816950719

**Comment by csagan5** [comment]
@uazo or is this rather a duplicate of #1047? In your comment of #1047 it looks like there is a different crash.

A few questions:
1. can we reproduce this bug?
2. since it is not reproducible in Chromium, can we identify which patch is responsible? For example by building Bromite without possibly-related patches until we build a version where bug is not reproducible.

**Comment by uazo** [comment]
> can we reproduce this bug?

yes

> can we identify which patch is responsible? 

yes, always incognito.

but the problem is how to solve it.
in my opinion it is to do a revert of my latest changes and simply remove the histogram: the code seems to be correct, if we want the incognito PWA with the "always incognito" flag active.
I'm trying.

**Comment by csagan5** [comment]
Fixed in `90.0.4430.204`.
